packages/entities/entities-plugins/src/definitions/metadata.ts

```diff
@@ -767,6 +767,12 @@ export const CREDENTIAL_SCHEMAS: Record<string, any> = {
  * @returns URL for the plugin icon
  */
 export const getPluginIconURL = (name: string) => {
-  const imageName = PLUGIN_METADATA[name]?.imageName || name
-  return new URL(`../assets/images/plugin-icons/${imageName}.png`, import.meta.url).href
+  const imageName = PLUGIN_METADATA[name]?.imageName || name // default icon is 'missing'
+  const iconURL = new URL(`../assets/images/plugin-icons/${imageName}.png`, import.meta.url).href
+
+  if (iconURL.endsWith('/undefined.png') || iconURL.endsWith('/undefined')) {
```

Comment:
question: why not use a more generic `iconURL.includes('undefined)` instead of checking for two separate `endsWith` conditions?

Reply:
We can have an icon that might have the word undefined in it, e.g. `user-undefined`

Reply:
How about this instead then?
```suggestion
  if (iconURL.includes('plugin-icons/undefined')) {
```

Reply:
Yeah, this should work for now until we allow custom asset directory as an arg to the method.

Reply:
I think just checking for `undefined` in the string is sufficient; we aren't going to have icons with `*undefined*` in the name